### PR TITLE
Update CICD for Building Docker Image for CoWork Platform

### PR DESCRIPTION
**Pull request by huma0605**
## Proposed changes (including videos or screenshots)

This PR updates the CI workflow to create docker image and upload to our private DockerHub repo.

Now every PR will create a docker image with tag `pr#.preview`. And with release or into main, it will create a docker image with tag `official`.

### CODE REVIEW

**Comment by huma0605** [comment]
publish docker image

**Comment by huma0605** [comment]
publish docker image

File: .github/actions/build-docker-image-dockerhub/action.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    required: false
+
+outputs:
+  image-name:
+    value: ${{ steps.build-image.outputs.image-name }}
+
+runs:
+  using: composite
+  steps:
+    # - shell: bash
```
Comment (hsinyenchung1) [inline .github/actions/build-docker-image-dockerhub/action.yml]: @huma0605 do we want to keep this? i'm curious why we command it out

File: apps/meteor/app/lib/server/functions/saveUser.js [hunk 2]
```diff
@@ -421,6 +421,11 @@ export const saveUser = async function (userId, userData) {
 		updateUser.$set['emails.0.verified'] = userData.verified;
 	}
 
+	if (!updateUser.$set.customFields) {
+		updateUser.$set.customFields = {};
+	}
+	updateUser.$set.customFields.messageToAgents = {};
```
Comment (hsinyenchung1) [inline apps/meteor/app/lib/server/functions/saveUser.js]: do we also want to create the messageToAgents property when we create a new user? or saveUser is fine?

Comment (hsinyenchung1) [reply apps/meteor/app/lib/server/functions/saveUser.js]: if i understand it correctly, we want to set the messageToAgents to empty object when saveUser is invoked? why do we want to set to {} everytime the saveUser is called?

File: .github/actions/build-docker-image-dockerhub/action.yml (hunk 1 above)
Comment (huma0605) [reply .github/actions/build-docker-image-dockerhub/action.yml]: yeah, let me remove them.

File: apps/meteor/app/lib/server/functions/saveUser.js (hunk 2 above)
Comment (huma0605) [reply apps/meteor/app/lib/server/functions/saveUser.js]: I removed them. We might do this on the CoWorkBot side.

**Review by hsinyenchung1 — APPROVED** [review]

**Pull request merged**